airflow/operators/python.py

```diff
@@ -333,20 +337,31 @@ def __init__(
             templates_exts=templates_exts,
             **kwargs,
         )
-        self.requirements = list(requirements or [])
+        if not isinstance(requirements, str):
+            self.requirements = list(requirements or [])
+        else:
+            self.requirements = requirements
         self.string_args = string_args or []
         self.python_version = python_version
         self.use_dill = use_dill
         self.system_site_packages = system_site_packages
-        if not self.system_site_packages and self.use_dill and 'dill' not in self.requirements:
-            self.requirements.append('dill')
         self.pickling_library = dill if self.use_dill else pickle
 
+    def pre_execute(self, context: Any):
+        if isinstance(self.requirements, list):
+            return
+
+        import pkg_resources
+        self.requirements = [str(req) for req in pkg_resources.parse_requirements(self.requirements)]
```

Comment:
The requirements file format can do more than listing requirements, and those cannot be parsed by `parse_requirements` (I don’t remember whether setuptools would throw or just silently drop them.) So instead of trying to pre-parse the content, I would try to parse it to pip instead.

* Remove `pre_execute`
* Change `prepare_virtualenv` to accept `requirements` as str (plus the currently supported list form)
* When doing `pip install`, write the str type `requirements` to a temporary file for `pip intall -r`.

Reply:
Ack, makes sense, let me work on that change.

Reply:
@uranusjr I have tried an implementation of your proposed approach, can you once re-review?